### DISCUSSION TITLE
Changing Serial CDC name

### DISCUSSION THREAD

**Discussion by geraldjust** [stm32duino/Arduino_Core_STM32#1434] [Q&A] [answered]
Hi there , i was wondering if anyone has a way to change the "name" of the USB_PRODUCT_STRING from the built in ones to a custom string. 

Acording to  https://github.com/stm32duino/Arduino_Core_STM32/blob/master/cores/arduino/stm32/usb/usbd_desc.c
line #67 looks like if i define it it will change it but i have not been able to do that. I can easily change the VID and PID via boards.txt by adding `build.vid=0x239A` as a example.
But was wondering if there's anything similar like that compiler flag to set USB_PRODUCT_STRING to a custom string. 

Thanks!

**Comment by fpistm** [comment, marked as answer]
Hi @geraldjust 
you can try to use the `build_opt.h`: https://github.com/stm32duino/wiki/wiki/Customize-build-options-using-build_opt.h

`-DUSB_PRODUCT_STRING="custom  USB"`
